osv2/index.go

```diff
@@ -0,0 +1,152 @@
+package osv2
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+
+	"github.com/opensearch-project/opensearch-go/v2"
+	"github.com/opensearch-project/opensearch-go/v2/opensearchapi"
+
+	"github.com/CrowdStrike/opensearchtools"
+)
+
+type IndexRequest struct {
+	Operation opensearchtools.IndexOperation
+	Routing   string
+}
+
+// NewIndexRequest instantiates an empty IndexRequest
+func NewIndexRequest() *IndexRequest {
+	return &IndexRequest{}
+}
+
+// Add an index operation to the IndexRequest.
+func (i *IndexRequest) Add(operation opensearchtools.IndexOperation) *IndexRequest {
+	i.Operation = operation
+	return i
+}
+
+func (i *IndexRequest) WithRouting(routing string) *IndexRequest {
+	i.Routing = routing
+	return i
+}
+
+// Validate the index request based on the types
+func (i *IndexRequest) Validate() error {
+	switch i.Operation.OperationType() {
+	case opensearchtools.IndexCreate:
+		if len(i.Operation.GetIndices()) > 1 {
+			return fmt.Errorf("the index length for create should be 1, current lenght: %d", len(i.Operation.GetIndices()))
+		}
+	case opensearchtools.IndexGet:
+		// implement some checks
+	case opensearchtools.IndexExists:
+		// implement some checks
+	case opensearchtools.IndexDelete:
+		// implement some checks
+	}
+	return nil
+}
+
+func (i *IndexRequest) Do(ctx context.Context, client *opensearch.Client) (*opensearchtools.OpenSearchResponse[IndexResponse], error) {
+	if err := i.Validate(); err != nil {
+		return nil, err
+	}
+	rawBody, jErr := i.ToOpenSearchJSON()
+	if jErr != nil {
+		return nil, jErr
+	}
+	var osResp *opensearchapi.Response
+	var rErr error
+	switch i.Operation.OperationType() {
```

Comment:
To be honest, I'm not a big fan of having 1 request that does 4 different things.  I'd prefer to see these split into `CreateIndexRequest, `DeleteIndexRequest`, etc.